pkgs/applications/networking/instant-messengers/mirage/default.nix

```
@@ -0,0 +1,54 @@
+{ stdenv, lib, mkDerivation, fetchFromGitHub
+, qmake, pkgconfig, olm, wrapQtAppsHook
+, qtbase, qtquickcontrols2, qtkeychain, qtmultimedia, qttools, qtgraphicaleffects
+, python3Packages, pyotherside
+}:
+
+let pypkgs= with python3Packages; [
+  aiofiles filetype matrix-nio appdirs cairosvg
+  pymediainfo setuptools html-sanitizer mistune blist
+  pyotherside
+]; in
+mkDerivation rec {
+  pname = "mirage";
+  version = "0.5.2";
+
+  src = fetchFromGitHub {
+    owner = "mirukana";
+    repo = pname;
+    rev = "v${version}";
+    sha256 = "0i891fafdncdz1xg6nji80jb86agsrbdvai9nwf1yy126q7piryv";
+    fetchSubmodules = true;
+  };
+
+  nativeBuildInputs = [ pkgconfig qmake wrapQtAppsHook python3Packages.wrapPython ];
+
+  buildInputs = [
+    qtbase qtmultimedia
+    qtquickcontrols2
+    qtkeychain qtgraphicaleffects
+    olm pyotherside
+  ];
+
+  propagatedBuildInputs = pypkgs;
+
+  pythonPath = pypkgs;
+
+  qmakeFlags = [ "PREFIX=${placeholder "out"}" ];
+
+  postInstall = ''
+    buildPythonPath "$out $pythonPath"
+    wrapProgram $out/bin/mirage \
+      --prefix PYTHONPATH : "$PYTHONPATH" \
+      --prefix QML2_IMPORT_PATH : "${pyotherside}/${qtbase.qtQmlPrefix}"
+  '';
```

to avoid double wrapping
```suggestion
  dontWrapQtApps = true;
  postInstall = ''
    buildPythonPath "$out $pythonPath"
    wrapProgram $out/bin/mirage \
      --prefix PYTHONPATH : "$PYTHONPATH" \
      --prefix QML2_IMPORT_PATH : "${pyotherside}/${qtbase.qtQmlPrefix}" \
      ''${qtWrapperArgs[@]}
  '';
```

---

If I do this, I get `! 13:15:58 | QQmlComponent: Component is not ready` as the only console output.

I tried, along with `dontWrapQtApps`, adding either `''${qtWrapperArgs[@]}"` or `"''${qtWrapperArgs[@]}"` and neither worked.

---

did you put the `\` and the end of the previous line? I was able to get this to build and launch fine.

---

also, it should just be `''${qtWrapperArgs[@]}` as I've written. Want it to be expanded as part of the arguments passed to `wrapProgram`

The `''$` is to prevent nix from interpolating it, and so it should just expand as a bash array.

---

:( No `set -u` for the hooks I guess. I think I had spelled `wrapper` wrong...

---

(`"''${qtWrapperArgs[@]}"` appears in nixpkgs tree many times and expands correctly, despite how it looks. also, it works :grin: )

---

I'm just not quite sure why it does work.... the double quotes should be making it into a single "string". w/e, if it works, it works